lib/reqTemplate.js

```diff
@@ -0,0 +1,132 @@
+"use strict";
+
+var URI = require('swagger-router').URI;
+var rbUtil = require('./rbUtil');
+var TAssembly = require('tassembly');
+
+/**
+ * Creates a function that sets a value at the path and initializes all objects along the path.
+ *
+ * @param path the path within an object separated with dots {e.g. test.body.a.b.c}
+ * @returns {Function} the function to set a value at path and initialize object along the path if needed
+ * @private
+ */
+function _setAtPath(path) {
+    var refArr = path.split('.').map(function(elem) {
+        return '["' + elem + '"]';
+    });
+    var code =  'if (value !== undefined) {\n';
+    for (var i = 1; i < refArr.length; i++) {
+        var ref = refArr.slice(0, i).join('');
+        code += 'obj' + ref + ' = obj' + ref + ' || {};\n';
+    }
+    code += 'obj' + refArr.join('') + ' = value;\n';
+    code += '}';
+    /* jslint evil: true */
+    return new Function('obj', 'value', code);
+}
+
+/**
+ *  Constructs a list of resolvers for all templates in a request spec
+ *
+ * @param subspec a request spec or subspec for a part of request
+ * @param reqPart name of request part (e.g. headers, body, query)
+ * @param path path to the current subspec within a full spec
+ * @returns {Array} an array of template resolvers for every template found in the spec.
+ *          Resolvers are functions with (newRequest, request) arguments that modify newRequest param.
+ * @private
+ */
+function _createTemplateResolvers(subspec, reqPart, path) {
+
+    /**
+     * Makes an array of resolvers for every template found within a template spec object.
+     *
+     * @param templateSpec A subspec of an original request spec.
+     * @param templatePath A path to the current subspec
+     * @returns {Array} and array of resolvers for each template found in the subspec
+     */
+    function makeResolvers(templateSpec, templatePath) {
+        var setValue = _setAtPath(templatePath);
+        if (templateSpec instanceof Object) {
+            return _createTemplateResolvers(templateSpec, reqPart, templatePath);
+        } else if (/^\{[^}]+}$/.test(templateSpec)) {
+            var template = /^\{([^}]+)}$/.exec(templateSpec)[1];
+            var buildModel;
+
+            if (!/^(\$\.)?([a-zA-Z][a-zA-Z0-9-]*)(\.[a-zA-Z][a-zA-Z0-9-]*)*$/.test(template)) {
+                throw new Error('Invalid template ' + template);
+            }
+
+            if (template.indexOf('$.') === 0) {
```

Comment:
`template[0] === '$'` should be faster (as we know from the regex above no other combo with `$` is allowed)